### PR TITLE
voting function displays index #, but not the value, aka normal, crit…

### PR DESCRIPTION
**Pull request by MatthewArvidson**
Our code at present has linked the importanceArray to local storage and we can increment and decrement the values and they persist in localStorage. However we are getting it back as the # of the impotanceArray, and not the associated value, aka none, low, normal, high, critical.

### CODE REVIEW

**Comment by selfup** [comment]
Well as you loop through localStorage you can reference this array by `index` to pull the correct value out

You will also need to store the `importanceArray` in your constructor:

```javascript
this.importanceArray = ['none', 'low', 'normal', 'high', 'critical'];
```

Ex:

```javascript
var importanceArray = this.importanceArray;

ideas.forEach(function (idea, index) {
  console.log(idea.title);
  console.log(importanceArray[index]);
});
```

That way you can get the correct value per node :smile:

***

Another option is to just store the value in an Idea object by getting the array position of the Idea.

```html
<!-- If a node were to be created with a `data-nodeIndex` -->
```

```javascript
var ideas = [
  { title: 'wow', body: 'lol', importance: this.defaultImportance },
];

// then query dom, get idea index from dom node once you find it

var index = node.data('nodeIndex');

ideas[index].quality = newQuality;
```

**Review by MatthewArvidson — COMMENTED** [review]
Voting function is now persisting and the index of the array is displaying the value not the #.

**Review by patrickjneel — APPROVED** [review]

File: script.js [hunk 1]
```diff
@@ -46,7 +69,9 @@ function Card(content) {
 this.title = content.title;
 this.task = content.task;
 this.id = content.id || Date.now();
-this.importanceIndex = content.importanceIndex || 0;
+this.importanceIndex = 2;
+this.importanceArray = ['none', 'low', 'normal', 'high', 'critical'];
```
Comment (selfup) [inline script.js:73]: Sweet seems to have helped!

File: script.js [hunk 2]
```diff
@@ -104,61 +136,70 @@ if (event.keyCode === 13) {
...
-  var articleElement = $(event.target).closest('article');
-  var id = articleElement.prop('id');
-  var card = Card.find(id);
-  var uniqueTodo = JSON.parse(localStorage.getItem(id));
-  articleElement.find('.todo-importance').text(card.getImportance());
-  card.decrementImportance();
-  localStorage.setItem(id, JSON.stringify(uniqueTodo));
-  card.save();
+  var articleId = $(this).closest('article').prop('id');
+  var parsedObject = find(articleId);
```
Comment (selfup) [inline script.js:152]: ahh this seems to be the equivalent of `var todo`

Nice 👍

**Comment by selfup** [comment]
Glad you all figured it out 🎉

**Pull request closed without merging**